### PR TITLE
CHANGE(keystone): Split memcached group

### PR DESCRIPTION
**Pull request by fatpat**
##### SUMMARY

Currently, memcached is deployed on groups `oioswift` and `keystone`
without the possibility to deploy them elsewhere

Like the redis group, the memcached group can now be split into two
groups:

    memcached_keystone for keystone auth cache
    memcached_swift for swift cache

 ##### IMPACT
N/A

 ##### ADDITIONAL INFORMATION

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -10,8 +10,14 @@ openio_keystone_bind_address: "{{ openio_bind_address \
 
 openio_keystone_repo_managed: false
 openio_keystone_openstack_distribution: queens
-openio_keystone_nodes_group: keystone
 openio_keystone_db_nodes_group: galera
+openio_keystone_memcached_nodes_group: "{%- if 'memcached-keystone' in groups -%}
```
Comment (cdelgehier) [inline defaults/main.yml]: Don't use dashes

File: defaults/main.yml [hunk 2]
```diff
@@ -10,8 +10,14 @@ openio_keystone_bind_address: "{{ openio_bind_address \
...
+  memcached-keystone
+  {%- elif 'memcached' in groups -%}
+  memcached
+  {%- else -%}
+  oioswift
```
Comment (cdelgehier) [inline defaults/main.yml]: I don't understand this one

**Comment by cdelgehier** [comment]
Please fix the credentials.yml and fernet.yml too

File: defaults/main.yml (hunk 2 above)
Comment (vincent-legoll) [reply defaults/main.yml]: I think this is not strictly needed, but is the second fallback, and looks harmless

File: defaults/main.yml (hunk 1 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

File: defaults/main.yml (hunk 2 above)
Comment (fatpat) [reply defaults/main.yml]: deploy memcached in the follwing group with order preference:
- `memcached_keystone`
- `memcached` if `memcached_keystone` is not defined
- `oioswift` if neither `memcached` or `memcached_keystone` is defined

Comment (fatpat) [reply defaults/main.yml]: I don't really understand the missunderstanding on this one :) this is very straight forward

**Comment by fatpat** [comment]
> Please fix the credentials.yml and fernet.yml too

@cdelgehier 
what is there to fix in those files ?

**Comment by fatpat** [comment]
fixed in last commit

**Pull request merged**